pkg/psmdb/client.go

```diff
@@ -22,6 +22,13 @@ func MongoClient(ctx context.Context, k8sclient client.Client, cr *api.PerconaSe
 		return nil, errors.Wrapf(err, "get pods list for replset %s", rs.Name)
 	}
 
+	if len(pods.Items) == 0 {
+		pods, err = GetOutdatedRSPods(ctx, k8sclient, cr, rs.Name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "get outdated pods list for replset %s", rs.Name)
+		}
+	}
```

Comment:
For example, in `checkIfPossibleToRemove` function we are trying to connect to the replset which was deleted from the cr. We need to use `GetOutdatedRSPods` because it doesn't truncate list of pods. Please check comments: https://github.com/percona/percona-server-mongodb-operator/pull/1329/commits/dc975e48328a4f701b39229c28227327e1fff78f